raster-foundry/raster-foundry
# Support shapefile annotation upload

## Overview

This PR makes it possible to POST a shapefile to `projects/<id>/annotations/shapefile` and
create annotations from it

### Checklist

- [x] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated (raster-foundry/raster-foundry-api-spec#27)

### Notes

We need to standardize on what we expect for shapefile fields. In principle, a user should be able to
export a shapefile of annotations and re-import that same shapefile. But shapefile field names have a
ten character limit, so we can't just dump the fields. In export, we rename `description`
to `desc`, which we don't do in `geojson` export, so I'm not sure what we should pick there (or what
we should do about `machineGenerated`). Here, I `ogr2ogr`'ed the geojson export and used the
truncated fields names.

## Testing Instructions

 * `truncate table annotations;` (not necessary but makes it easier to see your new ones)
 * export some annotations as geojson from the ui (since shapefile export is broken), ideally
   from a prediction so they'll have confidence
 * `ogr2ogr -f 'ESRI Shapefile' annotations.shp annotations.geojson`
 * zip up all of the annotation shapefile files
 * upload your zipped annotation shapefile through the frontend

Closes #3534
